searchInRotatedArr.js

```diff
@@ -0,0 +1,63 @@
+// This file contains solution to the following problem:
+// https://leetcode.com/problems/search-in-rotated-sorted-array/
+
+const search = function (nums, target) {
+  let pointer1 = 0;
+  let pointer2 = nums.length - 1;
+  if (nums[pointer1] === target) {
+    return pointer1;
+  }
+  if (nums[pointer2] === target) {
+    return pointer2;
+  }
+  let left = nums;
+  let right = [];
+  while (nums[pointer1] - nums[pointer2] > 0) {
+    pointer1 += 1;
+    pointer2 -= 1;
+  }
+
+  if (nums[pointer1] > nums[pointer1 - 1] && nums[pointer1] < nums[pointer2]) {
+    pointer2 += 1;
+    left = nums.slice(0, pointer2);
+    right = nums.slice(pointer2);
+  }
+  if (
+    (nums[pointer1] < nums[pointer1 - 1] && nums[pointer1] < nums[pointer2]) ||
+    (pointer1 === pointer2 && nums[pointer1] < nums[pointer1 + 1])
+  ) {
+    left = nums.slice(0, pointer1);
+    right = nums.slice(pointer1);
+  }
+  if (pointer1 > pointer2) {
+    left = nums.slice(0, pointer1);
+    right = nums.slice(pointer2 + 1);
+  }
+  if (pointer1 === pointer2 && nums[pointer1] > nums[pointer1 + 1]) {
+    left = nums.slice(0, pointer1 + 1);
+    right = nums.slice(pointer1 + 1);
+  }
+
+  const searchTarget = (arr) => {
```

Comment:
Lets pass in `target` as argument into `searchTarget` (similar to how `arr` is an argument there) to be consistent. 

And indeed it's better to pass those explicitly as arguments (like in functional paradigm) not implicitly reference them just because we can access outer scope.